dbms/src/Storages/DeltaMerge/ReadThread/SegmentReadTaskScheduler.cpp

```diff
@@ -39,32 +39,63 @@ SegmentReadTaskScheduler::~SegmentReadTaskScheduler()
 }
 
 void SegmentReadTaskScheduler::add(const SegmentReadTaskPoolPtr & pool)
+{
+    std::lock_guard lock(add_mtx);
```

Comment:
Do we still need this guard on `add_mtx`? Seems that `submitPendingPool` is thread-safe

Reply:
Please add comments like `To avoid schedule from always failing to acquire the pending_mtx`